### PR TITLE
Added link to documentation for path error message on old osx sdk

### PR DESCRIPTION
**Pull request by JohanMabille**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/patches/0004-custom-error-message-for-old-sdk.patch [hunk 1]
```diff
@@ -1,14 +1,5 @@
-From 494752e4a46bdeb17cc044b7adbccab0ec39f126 Mon Sep 17 00:00:00 2001
-From: Isuru Fernando <isuruf@gmail.com>
-Date: Wed, 29 Dec 2021 19:28:35 -0800
-Subject: [PATCH 4/4] custom error message for old sdk
-
----
- libcxx/include/__availability | 8 +++++---
- 1 file changed, 5 insertions(+), 3 deletions(-)
-
```
Comment (isuruf) [inline recipe/patches/0004-custom-error-message-for-old-sdk.patch:9]: Please keep the From: line

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by isuruf** [comment]
Maybe we can add a test like `cat libcxx/include/__availability | grep "availability(macosx" | grep -v "conda-forge"` to make sure we the patch is up-to-date.

**Comment by h-vetinari** [comment]
> Maybe we can add a test like `cat libcxx/include/__availability | grep "availability(macosx" | grep -v "conda-forge"` to make sure we the patch is up-to-date.

Since there are several occurrences in that file, we'd have to negate the last condition and make sure there are none _without_ the link.

**Comment by h-vetinari** [comment]
Ah, sorry, had overlooked your "-v" (was looking up myself just now how to negate this...)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -62,13 +62,17 @@ outputs:
...
+        - libcxx/include/__availability
       commands:
         # Dummy var for different hashes
         - echo {{ dummy_var }}                                    # [osx]
         - if not exist %LIBRARY_INC%\\c++\\v1\\iterator exit 1    # [win]
         - test -f $PREFIX/include/c++/v1/iterator                 # [unix]
         - bash compile_test.sh                                    # [unix]
         - compile_test.bat                                        # [win]
+        # make sure we inserted link to our docs wherever it is necessary
+        - cat libcxx/include/__availability | grep "availability(macosx" | grep -v "conda-forge"  # [osx]
```
Comment (isuruf) [inline recipe/meta.yaml:75]:
```suggestion
        # make sure we inserted link to our docs wherever it is necessary
        - cat ${PREFIX}/include/c++/v1/__availability | grep "availability(macosx" | (! grep -v "conda-forge")  # [osx]
```

Comment (h-vetinari) [reply recipe/meta.yaml:75]: I thought about using the installed files, but I reckon it's exactly equivalent if we test the patched source file or where it ends up getting copied.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -62,13 +62,17 @@ outputs:
         - test_sources
         - compile_test.sh   # [unix]
         - compile_test.bat  # [win]
+      source_files:
+        - libcxx/include/__availability
```
Comment (h-vetinari) [inline recipe/meta.yaml:66]:
```suggestion
```

**Comment by isuruf** [comment]
Looks like we have 6 more places where we need to add this message.

**Comment by isuruf** [comment]
```
+ cat /Users/runner/miniforge3/conda-bld/libcxx_and_libcxxabi_1661454209728/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/include/c++/v1/__availability
+ grep 'availability(macosx'
+ grep -v conda-forge
        __attribute__((availability(macosx,strict,introduced=10.12)))           \
        __attribute__((availability(macosx,strict,introduced=10.9)))            \
        __attribute__((availability(macosx,strict,introduced=10.9)))            \
        __attribute__((availability(macosx,strict,introduced=10.9)))            \
        __attribute__((availability(macosx,strict,introduced=10.15)))           \
        __attribute__((availability(macosx,strict,introduced=11.0)))            \
```

**Comment by isuruf** [comment]
Maybe not the 10.9 places. We can ignore those by adding a `grep -v 10.9`

**Comment by h-vetinari** [comment]
Something is still off with the test, but the patch should be fine now. I need to log off for the weekend though (leaving for a wedding in a few hours), please don't wait for me.

**Pull request merged**